### DISCUSSION TITLE
[Proposal]: Conditional Assign

### DISCUSSION THREAD

**Discussion by sajjadarashhh** [dotnet/csharplang#7063] [Language Ideas] [answered]
a something we can use like ??= for another operations like => { ==,>,<,>=,<=, and etc)
that we can use them with following instruction:
int a= 5;
a ?> 4  =5;
or 
a ?>=5 = 10;
this can help in many projects

**Comment by svick** [comment]
It's not at all clear to me what is the proposed syntax supposed to mean. Which is why I think it's probably not a good idea to add it to the language.

**Reply by sajjadarashhh** [reply]
I think that we dont need a if statement or full inline statement sometimes
Assume we have model that name is car
Car have a model name
Now we want to fill null model names with this "not selected" and if that have a name use previous name
I wana make a feature for this to handle it without current features with simpler syntax like this:
Cars.select(item=>{item.ModelName?==null=>"NotSelected"; return item;});
I need to think about that syntax much more... To found best way
Now i have a one question
Is good or not?

**Reply by svick** [reply]
Is the `??` operator what you're looking for? For example `item.ModelName ?? "Not selected"` is basically equivalent to `item.ModelName != null ? item.ModelName : "Not selected"`.

**Comment by DavidArno** [comment, marked as answer]
> ... I need to think about that syntax much more... To found best way ...

I think that would be the wrong thing to do here. It gets referred to as [bike-shedding](https://en.wikipedia.org/wiki/Law_of_triviality). The precise syntax often will come last after much discussion, experimentation with the functionality etc.

What's important is that you clearly explain what it is that you are proposing. Show some examples of existing code, explain why it's a problem and then show the same functionality with some new example syntax. Do not waste time on trying to make that new syntax into anything fancy though.
